core/emitter.js

```diff
@@ -30,8 +29,10 @@ class Emitter extends EventEmitter {
   }
 
   handleDOM(event, ...args) {
+    const target = (event.composedPath ? event.composedPath()[0] : event.target);
```

Comment:
Why is this needed?

Reply:
If a listener exists above a shadow root, but the event fires from within said shadow root, the target will be the host of the shadow root rather than the element which fired the event.

`composedPath` will handle this, though it looks like support for it isn't well documented.